exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/spill/RecordBatchSizer.java

```diff
@@ -232,9 +251,8 @@ else if (width > 0) {
     }
   }
 
-  public static final int MAX_VECTOR_SIZE = ValueVector.MAX_BUFFER_SIZE; // 16 MiB
-
   private List<ColumnSize> columnSizes = new ArrayList<>();
+  private Map<String, ColumnSize> columnSizeMap = CaseInsensitiveMap.newHashMap();
```

Comment:
I will follow up on this on the mailing list. I was under the impression that Drill is case-**in**sensitive and did not support case sensitivity of columns in an external store like HBase or MapRDB. If Drill is case-sensitive then we have a bigger issue because some of the functional tests assume that it is case-**in**sensitive.

Reply:
Drill is case insensitive internally. The case insensitive map is correct. Thanks for catching this @ilooner!

Unfortunately, record batches have no name space: they are just a collection of vectors. So, we could end up with columns called both "c" and "C". This situation will case the column size map to end up with one entry for both columns, with the last writer winning.

The best solution would be to enforce name space uniqueness when creating vectors. The new "result set loader" does this, but I suspect other readers might not -- depending on the particular way that they create their vectors. Still, creating names that differ only in case is a bug and any code doing that should be fixed.

Reply:
Thanks for the explanation here and on the dev list @paul-rogers.